tests/node/adapter.test.ts

```diff
@@ -2,32 +2,120 @@ import type {
+  entities: Object.values(betterAuthEntities)
+})
 const randomUsers = createRandomUsersUtils(orm)
 
 const adapter = mikroOrmAdapter(orm, {
   debugLogs: {
     isRunningAdapterTests: true
   }
-})({})
+})
+
+suite("better-auth adapter tests", async () => {
+  const opts = {
+    user: {
+      fields: {
+        email: "email_address"
+      },
+      additionalFields: {
+        test: {
+          type: "string",
+          defaultValue: "test"
+        }
+      }
+    },
+    session: {
+      modelName: "sessions"
+    }
+  }
+
+  await runAdapterTest({
+    getAdapter: async (customOptions = {}) => {
+      const merged = merge(opts, customOptions)
+      const adapter = mikroOrmAdapter(baOrm, {
+        debugLogs: {
+          isRunningAdapterTests: true
+        }
+      })
+      return adapter(merged)
+    },
+    disableTests: {
+      // CREATE_MODEL: true,
+      // CREATE_MODEL_SHOULD_ALWAYS_RETURN_AN_ID: true,
+      // SHOULD_FIND_MANY: true,
+      // SHOULD_FIND_MANY_WITH_WHERE: true,
+      // SHOULD_FIND_MANY_WITH_OPERATORS: true,
+      // SHOULD_FIND_MANY_WITH_SORT_BY: true,
+      // SHOULD_FIND_MANY_WITH_LIMIT: true,
+      // SHOULD_FIND_MANY_WITH_OFFSET: true,
+      // SHOULD_UPDATE_WITH_MULTIPLE_WHERE: true,
+      // SHOULD_DELETE_MANY: true,
+      // SHOULD_NOT_THROW_ON_DELETE_RECORD_NOT_FOUND: true,
+      // SHOULD_NOT_THROW_ON_RECORD_NOT_FOUND: true,
+      // SHOULD_FIND_MANY_WITH_CONTAINS_OPERATOR: true,
+      // SHOULD_SEARCH_USERS_WITH_STARTS_WITH: true,
+      // SHOULD_SEARCH_USERS_WITH_ENDS_WITH: true,
+      // SHOULD_PREFER_GENERATE_ID_IF_PROVIDED: true,
+      // FIND_MODEL_WITH_SELECT: true,
+      // FIND_MODEL_WITH_MODIFIED_FIELD_NAME: true,
+      // UPDATE_MODEL: true,
+      // FIND_MODEL: true,
+      // FIND_MODEL_WITHOUT_ID: true,
+      // DELETE_MODEL: true
+      // SHOULD_WORK_WITH_REFERENCE_FIELDS: true
+    }
+  })
+})
 
-suite("create", () => {
+describe("Adapter Authentication Flow Tests", async () => {
+  const testUser = {
+    email: "test-email@email.com",
+    password: "password",
+    name: "Test Name"
+  }
+
+  const auth = betterAuth({
+    database: adapter,
+    emailAndPassword: {
+      enabled: true
+    }
+  })
+
+  afterAll(async () => {
+    await orm.getSchemaGenerator().refreshDatabase()
+  })
+
+  it("should successfully sign up a new user", async () => {
+    const user = await auth.api.signUpEmail({body: testUser})
+    expect(user).toBeDefined()
+  })
+
+  it("should successfully sign in an existing user", async () => {
+    const user = await auth.api.signInEmail({body: testUser})
+    expect(user.user).toBeDefined()
+  })
+})
+
+suite("create", async () => {
+  beforeEach(async () => await orm.getSchemaGenerator().refreshDatabase())
   test("a new record", async () => {
     const expected = randomUsers.createOne()
-    const actual = await adapter.create<UserInput, DatabaseUser>({
+    const actual = await adapter({}).create<UserInput, DatabaseUser>({
```

Comment:
No need to create fresh adapter here, you can create instance in the same test suite, if adapter created with the same parameters.